plugins/survey-resources/src/components/QuestionCollectionEditor.svelte

```diff
@@ -0,0 +1,185 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+
+<script lang="ts">
+  import survey from '../plugin'
+  import {
+    Button,
+    DropdownLabelsPopupIntl,
+    DropdownTextItem,
+    getEventPopupPositionElement,
+    IconAdd,
+    IconDelete,
+    IconDown,
+    IconUp,
+    Label,
+    showPopup
+  } from '@hcengineering/ui'
+  import { createQuery, getClient } from '@hcengineering/presentation'
+  import { Question, Survey } from '@hcengineering/survey'
+  import { Class, Ref, SortingOrder } from '@hcengineering/core'
+  import { getEditableQuestionDataClasses } from '../functions/getEditableQuestionDataClasses'
+  import { questionCreate } from '../functions/questionCreate'
+  import { questionInit } from '../functions/questionInit'
+  import QuestionEditor from './QuestionEditor.svelte'
+  import { questionDelete } from '../functions/questionDelete'
+  import { questionUpdate } from '../functions/questionUpdate'
+
+  export let object: Survey
+
+  const client = getClient()
+  const questionDataClassDropdownItems: DropdownTextItem[] = getEditableQuestionDataClasses(client).map((clazz) => ({
+    id: clazz._id,
+    label: clazz.label,
+    icon: clazz.icon
+  }))
+  const query = createQuery()
+
+  let questions: Question[] = []
+  let isMoving = false
+
+  $: if (object.questions > 0) {
+    query.query<Question>(
+      survey.class.Question,
+      { space: object.space, attachedTo: object._id },
+      (res) => {
+        questions = res
+      },
+      { sort: { rank: SortingOrder.Ascending } }
+    )
+  } else {
+    questions = []
+    query.unsubscribe()
+  }
+
+  async function onClickAdd (e: MouseEvent, index: number | null): Promise<void> {
+    showPopup(
+      DropdownLabelsPopupIntl,
+      {
+        items: questionDataClassDropdownItems,
+        selected: undefined
+      },
+      getEventPopupPositionElement(e),
+      async (dataClassRef?: Ref<Class<Question>>): Promise<void> => {
+        if (dataClassRef === undefined) {
+          return
+        }
+        const prevQuestion = index === null ? null : questions[index] ?? null
+        const nextQuestion = (index === null ? questions[0] : questions[index + 1]) ?? null
+        const question = questionInit(client, prevQuestion?.rank ?? null, nextQuestion?.rank ?? null, dataClassRef)
+        await questionCreate(client, object, question)
+      }
+    )
+  }
+
+  async function onClickMove (index: number, up: boolean): Promise<void> {
+    const question1 = questions[index]
+    const question2 = questions[index + (up ? -1 : 1)]
+
+    isMoving = true
+    const ops = client.apply(object._id)
+    await questionUpdate(ops, question1, { rank: question2.rank })
+    await questionUpdate(ops, question2, { rank: question1.rank })
+    await ops.commit()
+    isMoving = false
```

Comment:
when you moved a question, what question do you see under your cursor, the same (moved) or new question? 😄 it might be expected that when you move a question up, you scroll list of questions up as well, to keep the same question focused so you can press Up button multiple times.